tests/unit/test_client.py

```diff
@@ -138,6 +140,80 @@ async def test__get_metadata_with_psc(
     }
 
 
+async def test__get_metadata_error(
+    credentials: FakeCredentials,
+) -> None:
+    """
+    Test that AlloyDB API error messages are raised for _get_metadata.
+    """
+    # mock AlloyDB API calls with exceptions
+    client = AlloyDBClient(
+        alloydb_api_endpoint="https://alloydb.googleapis.com",
+        quota_project=None,
+        credentials=credentials,
+    )
+    get_url = "https://alloydb.googleapis.com/v1beta/projects/my-project/locations/my-region/clusters/my-cluster/instances/my-instance/connectionInfo"
+    resp_body = {
+        "error": {
+            "code": 403,
+            "message": "AlloyDB API has not been used in project 123456789 before or it is disabled",
+        }
+    }
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        with pytest.raises(ClientResponseError) as exc_info:
+            await client._get_metadata(
+                "my-project", "my-region", "my-cluster", "my-instance"
+            )
+        exc = exc_info.value
+        assert exc.status == 403
+        assert (
+            exc.message
+            == "AlloyDB API has not been used in project 123456789 before or it is disabled"
+        )
+    await client.close()
+
+
+async def test__get_metadata_error_parsing_json(
+    credentials: FakeCredentials,
+) -> None:
+    """
+    Test that AlloyDB API error messages are raised for _get_metadata when
+    response JSON fails to be parsed.
+    """
+    # mock AlloyDB API calls with exceptions
+    client = AlloyDBClient(
+        alloydb_api_endpoint="https://alloydb.googleapis.com",
+        quota_project=None,
+        credentials=credentials,
+    )
+    get_url = "https://alloydb.googleapis.com/v1beta/projects/my-project/locations/my-region/clusters/my-cluster/instances/my-instance/connectionInfo"
+    resp_body = ["error"]  # invalid json
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        with pytest.raises(ClientResponseError) as exc_info:
+            await client._get_metadata(
+                "my-project", "my-region", "my-cluster", "my-instance"
+            )
+        exc = exc_info.value
+        assert exc.status == 403
+        assert (
+            exc.message
```

Comment:
nit: no real value of doing `exc = exc_info.value`
```suggestion
        assert exc_info.value.status == 403
        assert (
            exc_info.value.message
```

Reply:
Done